### PR TITLE
Adds remote into created package.

### PR DESCRIPTION
**Pull request by pomek**
Fixes: #89 

During `gulp create-package`, remote was not created. Now, after init a repository, remote will be added and `git push` shouldn't throw any errors.

### CODE REVIEW

**Comment by Reinmar** [comment]
Please squash last commits. There shouldn't be ones like fixing variable names when that variable was introduced by previous commit.

File: dev/tasks/dev/utils/git.js [hunk 1]
```diff
@@ -142,5 +143,20 @@ module.exports = {
...
+	/**
+	 * Adds remote to repository under specified path.
+	 *
+	 * @param {String} pluginName
+	 * @param {String} repositoryPath
+	 */
+	addRemote( pluginName, repositoryPath ) {
+		const addRemoteCommands = [
+			`cd ${ repositoryPath }`,
+			`git remote add ${ defaultOrigin } git@github.com:ckeditor/${ pluginName }.git`
```
Comment (Reinmar) [inline dev/tasks/dev/utils/git.js]: GitHub url cannot be hardcoded. See the variables available in `create-package.js` - there's `gitHubUrl`.

**Comment by Reinmar** [comment]
I left a comment in the code about one minor issue.

**Comment by pomek** [comment]
@Reinmar, fixed. You can review one more time.

File: dev/tasks/dev/utils/git.js [hunk 2]
```diff
@@ -142,5 +143,20 @@ module.exports = {
...
+	/**
+	 * Adds remote to repository under specified path.
+	 *
+	 * @param {String} repositoryPath
+	 * @param {String} repositoryUrl
+	 */
+	addRemote( repositoryPath, repositoryUrl ) {
+		const addRemoteCommands = [
+			`cd ${ repositoryPath }`,
+			`git remote add ${ defaultOrigin } ${ repositoryUrl }`
```
Comment (Reinmar) [inline dev/tasks/dev/utils/git.js]: This is adding `ckeditor/ckeditor5-foo` as an origin. That's not a correct path.

**Comment by Reinmar** [comment]
Now the code adds an incorrect URL as a remote.

**Comment by pomek** [comment]
I did some fixes. Now, if a name will not contain a valid URL, `git.addRemote` will add string `https://github.com/` as a prefix for the name. Otherwise, I assume the name is a correct URL, so I will do nothing.

**Comment by Reinmar** [comment]
I don't understand. Why may the variable not contain the default value that I can see during creation of a new package? There's the default value coded somewhere and this method should return it: https://github.com/ckeditor/ckeditor5/pull/197/files#diff-f087c1ace7f9141506883a96952cf5fdR80

**Comment by pomek** [comment]
The variable `gitHubUrl` contains exactly the same string what user provided. It can be a correct URL, but it also can be some string (like default value in a configuration prompt - `ckeeditor/ckeeditor5-{package_name}`).

If I want to get a full URL - I have two ways:
1. Validate user's input - only correct URL should pass the validation,
2. Modify user's input - it still does not ensure that we'll build a correct URL.

**Comment by Reinmar** [comment]
GitHub URL contains what the user was asked for – the path on GitHub. It's not meant to contain a working Git URL. These are two totally different things. So you need to transform one into another.

**Comment by pomek** [comment]
I guess I [did it](https://github.com/ckeditor/ckeditor5/pull/197/files#diff-9c86580a845bd95e8a2694f275c4b70dR158).

File: dev/tasks/dev/utils/git.js [hunk 3]
```diff
@@ -142,5 +144,26 @@ module.exports = {
...
 		tools.shExec( commitCommands.join( ' && ' ) );
+	},
+
+	/**
+	 * Adds remote to repository under specified path.
+	 *
+	 * @param {String} repositoryPath
+	 * @param {String} repositoryUrl
+	 */
+	addRemote( repositoryPath, repositoryUrl ) {
```
Comment (Reinmar) [inline dev/tasks/dev/utils/git.js]: That's a wrong argument name. You get `gitHubUrl` and you turn it into `repositoryUrl`.

**Comment by Reinmar** [comment]
Not yet. The `addRemote()` method should accept only one type of param and it's `gitHubPath`. Right now it accepts `gitHubUrl` which is parsed to an url and a protocol is checked... this is unnecessary. There's no use case for that.

So:

1. Fix naming of `gitHubUrl` – it's not a URL, but a path. Only inside `addRemote()` it's extended to a full URL.
2. Remove the parsing, because there's no use case for that.
3. Remember about updating the inquiry text from `GitHub URL` to `GitHub path`.

**Comment by pomek** [comment]
> Remove the parsing, because there's no use case for that.

I'm not sure. Look on the pictures below:

image

image

It does not matter what user has provided. `gitHubUrl` can be a valid URL, but it can also be a not valid URL, but valid `gitHubPath`. 

`Create-package` tool won't throw any error. User can provide (as `gitHubUrl`) following data:
* `ckeditor/ckeditor5-package`,
* `https://github.com/ckeditor/ckeditor5-package`,
* `foo`

Since two ways are correct, creating package tool has to detect, what user has provided.

**Comment by Reinmar** [comment]
> Create-package tool won't throw an error. User can provide (as gitHubUrl) following data:
> ...
> And nothing will happen.

It's not a problem for this ticket. You need to apply YAGNI here. There was no report that anyone typed there an incorrect value. Especially that we're going to use the value that's suggested automatically.

So again, KISS :)

File: dev/tasks/dev/tasks/create-package.js [hunk 4]
```diff
@@ -79,7 +80,7 @@ module.exports = ( ckeditor5Path, workspaceRoot ) => {
 			return inquiries.getPackageGitHubUrl( packageName );
```
Comment (Reinmar) [inline dev/tasks/dev/tasks/create-package.js]: Here it's called `getPackageGitHubUrl` but below it's `gitHubPath`.

**Comment by Reinmar** [comment]
Still a small R- because of inconsistent naming.

File: dev/tasks/dev/tasks/create-package.js (hunk 4 above)
Comment (pomek) [inline dev/tasks/dev/tasks/create-package.js]: Ye, I was not sure whether I should also rename method or not…

Comment (Reinmar) [inline dev/tasks/dev/tasks/create-package.js]: Better do it now, once we realised that it's incorrect. Also the inquiry message should be changed as I already commented in https://github.com/ckeditor/ckeditor5/pull/197#issuecomment-225570029.

**Comment by Reinmar** [comment]
All fine now :).

**Pull request merged**